Goal: the function getlogin() and ttyname() can return NULL,
which is not checked. (Bastian Kleineidam - 29 Sep 2002)
Fixes: #162303

Status wrt upstream: already applied upstream.

Index: shadow-4.0.3/src/newgrp.c
===================================================================
--- shadow-4.0.3.orig/src/newgrp.c	2005-05-19 19:41:07.799079000 +0200
+++ shadow-4.0.3/src/newgrp.c	2005-05-19 20:24:06.819079000 +0200
@@ -386,8 +386,13 @@
 		SYSLOG ((LOG_INFO, "user `%s' switched to group `%s'",
 			 name, group));
 	if (getdef_bool ("SYSLOG_SG_ENAB")) {
-		char *loginname = xstrdup (getlogin ());
-		char *tty = xstrdup (ttyname (0));
+		char *loginname = getlogin ();
+		char *tty = ttyname (0);
+
+		if (loginname != NULL)
+		  loginname = xstrdup (loginname);
+		if (tty != NULL)
+		  tty = xstrdup (tty);
 
 		if (loginname == NULL)
 			loginname = "???";
